src/document.rs

```diff
@@ -559,6 +558,21 @@ impl Document {
 
         Ok(doc)
     }
+
+    /// Attempts to deserialize a `Document` from a byte stream.
+    pub fn from_reader<R: Read + ?Sized>(reader: &mut R) -> crate::de::Result<Document> {
+        Self::decode(reader, false)
+    }
+
+    /// Attempt to deserialize a `Document` that may contain invalid UTF-8 strings from a byte
+    /// stream.
+    ///
+    /// This is mainly useful when reading raw BSON returned from a MongoDB server, which
+    /// in rare cases can contain invalidly truncated strings (https://jira.mongodb.org/browse/SERVER-24007).
+    /// For most use cases, `Document::from_reader` can be used instead.
+    pub fn from_reader_utf8_lossy<R: Read + ?Sized>(reader: &mut R) -> crate::de::Result<Document> {
```

Comment:
If we anticipate there being more options to add here, maybe we should consider `from_reader_with_options`? Or do we think this will be it?

Reply:
I think it's fine to leave like this. If it becomes necessary, we can add a `with_options` variant in addition to these two.